src/main/java/org/justinbaur/bankteller/service/AdminServiceDatabaseImpl.java

```diff
@@ -1,29 +1,85 @@
 package org.justinbaur.bankteller.service;
 
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.justinbaur.bankteller.domain.Account;
+import org.justinbaur.bankteller.domain.Profile;
+import org.justinbaur.bankteller.domain.Address;
+import org.justinbaur.bankteller.domain.CustomerInfo;
 import org.justinbaur.bankteller.exceptions.AccountNotFound;
-import org.justinbaur.bankteller.repository.AccountRepository;
+import org.justinbaur.bankteller.exceptions.ProfileNotFound;
+import org.justinbaur.bankteller.repository.ProfileRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AdminServiceDatabaseImpl extends ProfileServiceImpl implements AdminService {
 
-public class AdminServiceDatabaseImpl implements AdminService {
+    private static final Logger LOG = LoggerFactory.getLogger(AdminServiceDatabaseImpl.class);
 
     @Autowired
-    AccountRepository repository;
-    
+    ProfileRepository repository;
+
     @Override
-    public void createAccount(Integer balance) {
-        // TODO Auto-generated method stub
+    public void createProfile(String firstName, String lastName, Address address) {
+
+        String id = firstName + lastName;
+
+        do {
+            id += (int) (Math.random() * 10000);
```

Comment:
Isn't this going to keep adding more and more numbers if its not found creating a different format than intended?  You'll have some Ids with super long numbers appened at some point

Reply:
Good catch, I should have added something to reset the ID on duplicate find.